cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/statements/ForStatement.kt

```diff
@@ -26,11 +26,13 @@
 package de.fraunhofer.aisec.cpg.graph.statements
 
 import de.fraunhofer.aisec.cpg.graph.AST
+import de.fraunhofer.aisec.cpg.graph.BranchingNode
+import de.fraunhofer.aisec.cpg.graph.Node
 import de.fraunhofer.aisec.cpg.graph.declarations.Declaration
 import de.fraunhofer.aisec.cpg.graph.statements.expressions.Expression
 import java.util.*
 
-class ForStatement : Statement() {
+class ForStatement : Statement(), BranchingNode {
```

Comment:
If we have the `ForEachStatement` as `StatementHolder` we should do that with every loop right? And switches as well?

Reply:
See above, this was introduced because of the fluent DSL and I will try to get rid of it.